### ISSUE TITLE
Twin game checker prevents undelete

### ISSUE THREAD

**Issue by nicktmro** [benini/scid#106]
**Summary**: When working on removing duplicates from a database, the Twin Checker Window prevents the user form unselecting a game for deletion and completely breaks the flow.

**Environment** 
SCID Version 5.0.0, Jan 27 2023 running on macOS Ventura, Using Tcl/Tk version: 8.6.13
Tcl(8.6.13),  TclOO(1.1.0),  Tk(8.6.13),  Ttk(8.6.13),  ezsmtp(1.0.0),  msgcat(1.6.1),  opt(0.4.8),  tcl::tommath(8.6.13),  tooltip(1.4.6),  ttk::theme::alt(8.6),  ttk::theme::aqua(8.6),  ttk::theme::clam(8.6),  ttk::theme::classic(8.6),  ttk::theme::default(8.6),  zlib(2.0.1).

Steps to reproduce:
1. Load a (large) database that contains duplicate games
2. From the Database Menu, select `Maintenance > Delete Games`
3. Set the filters and Click `Delete Games` ![Delete Twin Games filters](https://user-images.githubusercontent.com/603702/216704660-fed3bb3e-f13e-4b0f-9099-82d22532d034.png)
4. Click OK on the Popup Window the show the number of twin games that were found
5. SCID has now loaded the Twin Game Checker 
![Twin game checker is ready](https://user-images.githubusercontent.com/603702/216705353-c5d296f7-c626-4db5-ac0a-a90442410152.png)
6. The game on the right hand side is not selected for deletion (and this can also be verified in the Game List: only the game on the left hand side is flagged for deletion and appears greyed out)
7. Click the `Deleted` checkbox on the right hand side of the Twin Game Checker

Expected results:
- both left and right hand side games are still visible and are marked for deletion
- the Game List shows both games as marked for deletion

Actual results:
- both games are marked for deletion
- the Game List shows both games marked for deletion
- the Twin Game Checker "lost" the game on the right hand side and there is no way to Undo the deletion in the Twin Game Checker window. 
![Right hand side game is gone](https://user-images.githubusercontent.com/603702/216706162-998a78be-7ef5-4a89-9aa8-7fac7f197413.png)

Notes:
- clicking Previous / Next Pair does not help
- using the `u` undelete both option does not work, either, because it only undelete the game on the left hand side. The game on the right hand side is still flagged for deletion.
- clicking Undelete in the Game List still works to remove the flags

Note: after Step 4, I also expected the Game List to be updated to only show the filtered games, but that did not happen. Will be the subject of another Issue, unless addressed first.

**Comment by nicktmro** [comment]
Note that unselecting the `Deleted` checkbox on the left hand side also leads to the right hand side "twin" game disappearing from the Twin game checker dialog.

**Comment by benini** [comment]
Many thanks.
Fixed with:
https://github.com/benini/scid/commit/5aade37aabff6966fc7f1508f82fe8f5fb3fbd64

**Comment by nicktmro** [comment]
Thank you for fixing this so quickly. I just compiled and tested the commit, and the reported issue is gone. Using the mouse to toggle the checkbox works as expected.
Unfortunately, upon further testing, I found that using the `2` keyboard shortcut to `flip` the selection does not work.. 

Note: it is surprising because using `1` to select and unselect the left hand side game works as expected.
